ext/temporary/CRM/Temporary/Schema.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * Update temporal schema (TIMESTAMPs<=>DATETIMEs).
+ */
+class CRM_Temporary_Schema {
+
+  /**
+   * If the temporal schema requires any triggers, define them here.
+   *
+   * @param $info
+   * @param null $tableName
+   */
+  public static function createSqlTriggers(&$info, $tableName = NULL) {
+    if (temporary_timestamps() !== 'hybrid') {
+      return;
+    }
+
+    if ($tableName && !preg_match(TEMPORARY_TIMESTAMP_TABLES, $tableName)) {
+      // Optimization
+      return;
+    }
+
+    $tables = $tableName ? [$tableName] : static::walkTables();
+    foreach ($tables as $table => $daoClass) {
+      if (preg_match(TEMPORARY_TIMESTAMP_TABLES, $table)) {
+        foreach (static::walkFields([$table => $daoClass]) as $field) {
+          $tsField = $field['name'];
+          $gmtField = $field['name'] . '_gmt';
+
+          // The trigger-content varies based on whether before-trigger has good visibility into changes:
+          // - For fields that use `DEFAULT CURRENT_TIMESTAMP` and/or `ON UPDATE CURRENT_TIMESTAMP`, the before-trigger does
+          //   not have good visibility. It's sipler to just re-implement equivalent behavior.
+          // - For all other fields, do a conditional sync (DATETIME<=>TIMESTAMP).
+
+          if (mb_strpos($field['default'] ?? '', 'CURRENT_TIMESTAMP') !== FALSE) {
+            $onInsert = ["SET NEW.{$gmtField} = UTC_TIMESTAMP();"];
+          }
+          else {
+            $onInsert = [
+              "IF NEW.{$gmtField} IS NULL AND NEW.{$tsField} IS NOT NULL THEN",
+              "  SET NEW.{$gmtField} = CONVERT_TZ(NEW.{$tsField}, @@time_zone, '+0:00');",
+              "ELSEIF NEW.{$tsField} IS NULL AND NEW.{$gmtField} IS NOT NULL THEN",
+              "  SET NEW.{$tsField} = CONVERT_TZ(NEW.{$gmtField}, '+0:00', @@time_zone);",
+              "END IF;",
+            ];
+          }
+
+          if (mb_strtoupper($field['default'] ?? '') === 'CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP') {
+            $onUpdate = ["SET NEW.{$gmtField} = UTC_TIMESTAMP();"];
+          }
+          else {
+            $onUpdate = [
+              "IF NEW.{$tsField} <> OLD.{$tsField} THEN",
+              "  SET NEW.{$gmtField} = CONVERT_TZ(NEW.{$tsField}, @@time_zone, '+0:00');",
+              "ELSEIF NEW.{$gmtField} <> OLD.{$gmtField} THEN",
+              "  SET NEW.{$tsField} = CONVER_TZ(NEW.{$gmtField}, '+0:00', @@time_zone);",
```

Comment:
CONVER

Reply:
Doh. Confirmed that this typo caused an error. Fixed.